middleman-core/lib/middleman-core/renderers/haml.rb

```diff
@@ -52,6 +52,10 @@ def self.render_with_options(text, compiler_options)
               modified_options = options.dup
               modified_options[:context] = compiler_options[:context]
 
+              renderer_symbol = self.to_s.split(":").last.downcase.to_sym
+              renderer_options = compiler_options[:context].app.config[renderer_symbol]
```

Comment:
I was curious whether accessing the app's config through the context would be frowned upon but I did notice that the Sass renderer itself does something similar: https://github.com/olivierlacan/middleman/blob/9d2cf7b7a0366079958c9e1046d5cd1df8305268/middleman-core/lib/middleman-core/renderers/sass.rb#L90